crates/router/src/connector/trustpay/transformers.rs

```diff
@@ -0,0 +1,964 @@
+                    .change_context(errors::ConnectorError::FailedToObtainAuthType)?;
+                let amount =
+                    utils::to_currency_base_unit(item.request.amount, item.request.currency)?
+                        .parse::<f64>()
+                        .ok()
+                        .ok_or_else(|| errors::ConnectorError::RequestEncodingFailed)?;
+                Ok(Self {
+                    bank_refunds: Some(TrustpayRefundRequestBankRedirect {
+                        merchant_identification: MerchantIdentification {
+                            project_id: auth.project_id,
+                        },
+                        payment_information: BankPaymentInformation {
+                            amount: Amount {
+                                amount: format!("{:.2}", amount),
+                                currency: item.request.currency.to_string(),
+                            },
+                            references: References {
+                                merchant_reference: format!("{}_{}", item.payment_id, "1"),
+                            },
+                        },
+                    }),
+                    ..Default::default()
+                })
+            }
+            _ => Ok(Self {
+                card_refunds: Some(TrustpayRefundRequestCards {
+                    instance_id: item.request.connector_transaction_id.clone(),
+                    amount: utils::to_currency_base_unit(
+                        item.request.amount,
+                        item.request.currency,
+                    )?,
+                    currency: item.request.currency.to_string(),
+                    reference: item.payment_id.clone(),
+                }),
+                ..Default::default()
+            }),
+        }
+    }
+}
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct RefundResponse {
+    pub status: Option<i64>,
+    pub description: Option<String>,
+    pub instance_id: Option<String>,
+    pub payment_status: Option<String>,
+    pub payment_description: Option<String>,
+    pub result_info: Option<ResultInfo>,
+}
+
+impl<F> TryFrom<types::RefundsResponseRouterData<F, RefundResponse>>
+    for types::RefundsRouterData<F>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::RefundsResponseRouterData<F, RefundResponse>,
+    ) -> Result<Self, Self::Error> {
+        match (
+            item.response.payment_status,
+            item.response.instance_id,
+            item.response.result_info,
+        ) {
+            (Some(payment_status), Some(instance_id), _) => {
+                let (refund_status, msg) = get_refund_status(&payment_status)?;
+                let error = if msg.is_some() {
+                    Some(types::ErrorResponse {
+                        code: payment_status,
+                        message: msg.unwrap_or_default(),
```

Comment:
use consts::NO_ERROR_MESSAGE instead of default

Reply:
Done